relay-server/src/actors/spooler/mod.rs

```diff
@@ -490,6 +527,60 @@ impl OnDisk {
         }
     }
 
+    /// Unspools the entire contents of the on-disk spool.
+    async fn delete_and_fetch_all(
+        &mut self,
+        services: &Services,
+    ) -> Result<BTreeMap<QueueKey, Vec<ManagedEnvelope>>, BufferError> {
+        let mut result: BTreeMap<QueueKey, Vec<ManagedEnvelope>> = BTreeMap::new();
+
+        loop {
+            let mut envelopes =
+                sql::delete_and_fetch_all(BATCH_SIZE.min(self.buffer_guard.available() as i64))
+                    .fetch(&self.db)
+                    .peekable();
+            relay_statsd::metric!(counter(RelayCounters::BufferReads) += 1);
+
+            // Stream is empty, we can break the loop, since we read everything by now.
+            if Pin::new(&mut envelopes).peek().await.is_none() {
```

Comment:
Should be able to pin on the stack here via [pin!](https://doc.rust-lang.org/std/pin/macro.pin.html)

Reply:
done in [`b2b35f2`](https://github.com/getsentry/relay/pull/2949/commits/b2b35f2b50f796378f0b84da90c7061eaa9474b6)